lib/msf/ui/console/command_dispatcher/modules.rb

```diff
@@ -426,7 +438,19 @@ def cmd_search(*args)
               else
                 search_params = Msf::Modules::Metadata::Search.parse_search_string(match)
                 @module_search_results = Msf::Modules::Metadata::Cache.instance.find(search_params)
-              end
+
+                if sort and sort_options.include?(sort)
+                  if sort == 'date'
+                    sort='disclosure_date'
+                  end
+                  eval("@module_search_results.sort_by! {|meta| meta.#{sort}}")
```

Comment:
`eval()` statements are strongly discouraged as they can create security risks. Not sure why you can't just execute this as is without the `eval()` statement. Note that RuboCop also heavily warns against this so unless there is a compelling reason why there is no other option but to use `eval()` (which is pretty rare in my experience), we shouldn't be writing code like this.

Proposed solution is below but I haven't tested this yet. Will try test it after I finish reviewing.

```suggestion
                  @module_search_results.sort_by! {|meta| meta.send(sort)}
```

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/14917/commits/4382aeed28bccdfec455782e95fda9674f8490a4